src/realtime/presence/presence-entity-manager.ts

```diff
@@ -0,0 +1,46 @@
+import Entity from '../../entities/_base'
+import topics from '../../universe/topics'
+import { RealtimeClient } from '..'
+import PresenceHandler, { PresencePayload, PresenceStaffPayload } from './presence-handler'
+
+export default class PresenceEntityManager<T extends Entity<any, any>> {
+
+  constructor (
+    private readonly mqtt: RealtimeClient,
+    private readonly entity: T,
+    private readonly user: PresenceStaffPayload,
+    private readonly onPresenceUpdated: ((presence: PresenceStaffPayload[]) => void),
+    private readonly thresholdInSeconds: number,
+    private readonly relay: ((presence: PresencePayload) => void) | undefined = undefined,
+    protected presenceHandler: PresenceHandler | null = null) {
+      this.init()
+  }
+
+  public init (): void {
+    if (this.presenceHandler) {
```

Comment:
How about returning early?

Reply:
I think that branching is better if it's not a guard like if (x) return specially when there isn't much depth to the branching